src/Dhall/Import.hs

```diff
@@ -674,7 +676,10 @@ loadDynamic p = do
 
 -- | Load a `Path` as a \"static\" expression (with all imports resolved)
 loadStatic :: Path -> StateT Status IO (Expr Src X)
-loadStatic path = do
+loadStatic = loadStaticWith Dhall.Context.empty
+
+loadStaticWith :: Dhall.Context.Context (Expr Src X) -> Path -> StateT Status IO (Expr Src X)
+loadStaticWith ctx path = do
```

Comment:
I think you also need to change the recursive call to `loadStatic` inside this function to use `loadStaticWith`, so that the context is propagated to transitive imports